File: frontend/src/app/services/income/income.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

const BASIC_URL = environment['BASIC_URL'];

@Injectable({
  providedIn: 'root'
})
export class IncomeService {

  constructor(private http: HttpClient) { }

  postIncome(incomeDTO: any): Observable<any> {
    return this.http.post<[]>(BASIC_URL + "api/income", incomeDTO);
  }

  getAllIncomes(): Observable<any> {
    return this.http.get<[]>(BASIC_URL + "api/income/all");
  }

  getIncomeById(id: number): Observable<any> {
    return this.http.get<[]>(BASIC_URL + `api/income/${id}`);
  }

  updateIncome(id: number, incomeDTO: any): Observable<any> {
    return this.http.put<[]>(BASIC_URL + `api/income/${id}`, incomeDTO);
  }

  deleteIncome(id: number): Observable<any> {
    return this.http.delete<[]>(BASIC_URL + `api/income/${id}`);
  }

}

File: backend/src/main/java/com/yosephb/ExpenseTracker/services/income/IncomeService.java
package com.yosephb.ExpenseTracker.services.income;

import java.util.List;

import com.yosephb.ExpenseTracker.dto.IncomeDTO;
import com.yosephb.ExpenseTracker.entity.Income;

public interface IncomeService {
	
	Income postIncome(IncomeDTO incomeDTO);
	
	List<IncomeDTO> getAllIncomes();
	
	Income updateIncome(Long id, IncomeDTO incomeDTO);
	
	IncomeDTO getIncomeById(Long id);
	
	void deleteIncome(Long id);

}
